Automattic/wp-calypso
# Remove customizer: Add Full Site Editing (FSE) Active Status Check in Theme Enhancements

This PR introduces changes to the ThemeEnhancements component in the Site Settings of our application. The primary aim is to improve the user experience by making the interface more responsive to the Full Site Editing (FSE) active status.



Related to https://github.com/Automattic/wp-calypso/issues/71130#issuecomment-1576829569

## Proposed Changes

* Importing the withIsFSEActive helper from the calypso/data/themes/with-is-fse-active directory, allowing us to determine whether FSE is active or not.
* Adding isFSEActive and isFSEActiveLoading props to the component's properties to monitor the FSE active status and its loading state.
* Introducing conditional rendering logic in the JetpackModuleToggle and SupportInfo components, such that they will only render when the FSE is not active and is not loading. This change ensures that the CSS customization options are appropriately adjusted based on the FSE active status.
* Modifying the export of ThemeEnhancements to use the withIsFSEActive HOC, enhancing the connected component with FSE active status.

These changes aim to provide a more dynamic user experience, adjusting the UI based on the FSE active status. It ensures that the theme enhancement settings shown to the user are always relevant to their current setup.

## Testing Instructions



* Setup: Click on the Calypso live link provided in this PR.
* Visit Site Settings: Navigate to the Site Settings of a site.
* Access Writing Settings: Navigate to the Write submenu of the Settings menu.
* Locate Theme Enhancements Section: The enhanced CSS toggle shouldn't be visible if you have a block-based theme enabled, and should be visible if you aren't using a block based theme.
* Check UI Responsiveness: Check to see if the JetpackModuleToggle and SupportInfo components render correctly, depending on the active status of FSE. These components should only appear when FSE is not active.
* Check Console for Errors: While testing, keep your browser's console open to catch any possible errors or warnings.
* Repeat for simple, atomic, and JN sites. Sites without a paid Jetpack plan will not show the toggle regardless of the theme type.
## Pre-merge Checklist
